File: helpers/dbms.py
from pymongo import MongoClient
from .slackWrapper import get_message, post_to_channel

def post_to_slack(db):
  """ Helper method that posts a confession to every workplace"""
  all = db.find({})
  for workplace in all:
    msg = get_message(db, workplace['team_id'])
    if msg:
      print("Sending confessions")
      post_to_channel(db, workplace['team_id'], msg)


def clear_denied(db):
  """ Helper method that removes all the denied confessions """
  print("Clearing denied items")
  all = db.find({})
  for workplace in all:
    for index in range(len(workplace['messages']) - 1):
      if workplace['messages'][index]['denied'] or workplace['messages'][index]['posted']:
        del workplace['messages'][index]
    db.update_one({
        'team_id': workplace['team_id']
    }, {
        "$set": workplace
    })

File: helpers/dbms.js
const Confession = require('../models/confessions')
const {get_message, post_to_channel} = require("./slackWrapper")

async function post_to_slack() {
  // Helper method that posts a confession to every workplace
  const all = await Confession.find({})
  for (workplace of all) {
    const msg  = await get_message(workplace.team_id)
    if (msg) {
      console.log("sending confession")
      post_to_channel(workplace.team_id, msg)
    }
  }
}

async function clear_denied() {
  // Helper method that removes all the denied confessions
  const all = await Confession.find({})
  for (workplace of all) {
    for (index in workplace.messages) {
      let m = workplace.messages[index]
      if (m.denied || (m.approved && m.posted)) {
        workplace.messages.splice(index) // CHECK IF THIS WORKS!!!!
      }
    }
    Confession.findOneAndUpdate({team_id},
      {$set: {"messages": workplace.messages}},
      function(err, model) {
        if (err) console.log(err);
      }
    )
  }
}

module.exports = {post_to_slack, clear_denied}